azure-quantum/azure/quantum/target/ionq.py

```diff
@@ -54,6 +54,8 @@ def __init__(
         encoding: str = "",
         **kwargs
     ):
+        provider_id = kwargs.pop("provider_id", provider_id)
```

Comment:
I don't quite understand why we need to make each target support overriding the provider_id in the kwargs. I imagine we are only doing this to support the echo targets?